app/frontend/src/features/auth/signup/Signup.tsx

```diff
@@ -1,33 +1,133 @@
+import AuthBg from "../../../resources/auth-bg.png";
+import { loginRoute, signupRoute } from "../../../routes";
 import { useAuthContext } from "../AuthProvider";
 import CompleteSignupForm from "./CompleteSignupForm";
 import EmailForm from "./EmailForm";
 
+const useStyles = makeStyles((theme) => ({
+  signupPage: {
+    boxSizing: "border-box",
+    display: "flex",
+    flexDirection: "column",
+    alignItems: "center",
+    height: "100vh",
+    padding: `${theme.spacing(1)} ${theme.spacing(6)}`,
+  },
+  backgroundImage: {
+    position: "fixed",
+    left: 0,
+    right: 0,
+    zIndex: -1,
+    display: "block",
+    backgroundImage: `url(${AuthBg})`,
+    backgroundColor: "#f3f3f3",
+    backgroundPosition: "top center",
+    backgroundRepeat: "no-repeat",
+    backgroundSize: "contain",
+    height: "100vh",
+    filter: "blur(5px)",
+    opacity: "0.5",
+  },
+  divider: {
+    width: "100%",
+    marginTop: theme.spacing(2),
+    marginLeft: "auto",
+    marginRight: "auto",
+    borderTop: "1px solid #333333",
+  },
+  facebookButton: {
+    width: "100%",
+    height: "40px",
+    borderRadius: `${theme.shape.borderRadius * 2}px`,
+    boxShadow: "0px 0px 5px rgba(0, 0, 0, 0.25)",
+    minHeight: `calc(calc(${theme.typography.button.lineHeight} * ${
+      theme.typography.button.fontSize
+    }) + ${theme.typography.pxToRem(12)})`, // from padding
+    color: "#ffffff",
+    fontWeight: 700,
+    backgroundColor: "#2d4486",
+    marginTop: theme.spacing(6),
+    marginBottom: theme.spacing(2),
+  },
+  googleButton: {
+    width: "100%",
+    height: "40px",
+    borderRadius: `${theme.shape.borderRadius * 2}px`,
+    boxShadow: "0px 0px 5px rgba(0, 0, 0, 0.25)",
+    minHeight: `calc(calc(${theme.typography.button.lineHeight} * ${
+      theme.typography.button.fontSize
+    }) + ${theme.typography.pxToRem(12)})`, // from padding
+    color: "#ffffff",
+    fontWeight: 700,
+    backgroundColor: "#d9472e",
+  },
+  agreement: {
+    textAlign: "center",
+  },
+  logIn: {
+    marginTop: "auto",
+  },
+  logInLink: {
+    textDecoration: "none",
+    color: theme.palette.secondary.main,
+    fontWeight: 700,
+  },
+}));
+
 export default function Signup() {
-  const { authState } = useAuthContext();
+  const { authState, authActions } = useAuthContext();
   const authenticated = authState.authenticated;
   const error = authState.error;
+  const classes = useStyles();
+
+  useEffect(() => {
+    authActions.clearError();
+  }, [authActions]);
 
   return (
     <>
       {authenticated && <Redirect to="/" />}
-      <PageTitle>Signup</PageTitle>
-      {error && <Alert severity="error">{error}</Alert>}
-
-      <Switch>
-        <Route exact path={`${signupRoute}`}>
-          <EmailForm />
-        </Route>
-
-        <Route path={`${signupRoute}/:urlToken?`}>
-          <CompleteSignupForm />
-        </Route>
-      </Switch>
+      <Box className={classes.backgroundImage}></Box>{" "}
```

Comment:
Put the background image in `features/auth/resources` (if it's the same image for login and sign-up). I think the idea of the features folder is to keep related things as close as possible rather than organising by type of file/code.

Reply:
done